src/schema/v2/me/conversation/eventConnection/superCursor.ts

```diff
@@ -0,0 +1,62 @@
+import qs from "qs"
+import { URLSearchParams } from "url"
+
+type CursorState<T extends string> = { position: number } & Record<T, number>
+
+const PREFIX = "supercursor"
+const base64 = (str: string) => Buffer.from(str, "utf-8").toString("base64")
+const unBase64 = (str: string) => Buffer.from(str, "base64").toString("utf-8")
+
+export class SuperCursor<T extends string> {
```

Comment:
Mind adding a description and `@example` of what this does?

Reply:
Yep, but it is now covered by unit tests as well. In short it is an immutable object that contains a counter for each source you are querying.

Say you initialize a list of both bnmo artworks and auction lots from two different endpoints. You might call that with something like `fetchHybridConnection({ first: 10, after: 'asdfasdf' }, { lot: fetchSaleArtworks, bnmo: fetchBnmoArtworks })` and you will get back a connection of lots and sale artworks. _Internally_ this object is tracking 3 things: `position` which is the overall index of each element in total combined collection, plus `lot` and `bnmo` which is the **offset** within each individual connection.

If the cursor `asdfasdf` decodes to something like `offsets:position=9&lot=4&bnmo=6` (not sure about my index math there) it means that you are a) requesting page 2, after index 9 in the collection and b) requesting an offset of 4 and 6 respectively on each collection. In order to be certain we will have enough records we much get the 10 of each (the `first` arg) and then sort/slice.

Anyhow it's not an easy thing to explain on here, it's really something that requires the charlie day always sunny gif to fully communicate.